plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GroovyExtension.java

```diff
@@ -128,13 +128,13 @@ protected void setupTask(SpotlessTask task) {
 			}
 			//Add all Groovy files (may contain Java files as well)
 
-			UnionFileCollection union = new UnionFileCollection();
+			FileCollection union = getProject().files();
 			for (SourceSet sourceSet : convention.getSourceSets()) {
 				GroovySourceSet groovySourceSet = new DslObject(sourceSet).getConvention().getPlugin(GroovySourceSet.class);
 				if (excludeJava) {
-					union.add(groovySourceSet.getAllGroovy());
+					union = union.plus(groovySourceSet.getAllGroovy());
 				} else {
-					union.add(groovySourceSet.getGroovy());
+					union = union.plus(groovySourceSet.getGroovy());
 				}
```

Comment:
Just curious, is there a performance difference here?  Is `FileCollection.plus()` just as "lazy" as `UnionFileCollection`?

Reply:
The `FileCollection` returned by `getProject().files()` is immutable (as far as I can tell). The only way that a `UnionFileCollection` can be created using the public API is using this `plus` method.

So what's actually happening is that a bunch of `UnionFileCollection`'s are being unioned together.
Don't know if this is really a performance thing besides a bunch of short lived objects being created.

Reply:
Thanks!  Sounds good enough to me.